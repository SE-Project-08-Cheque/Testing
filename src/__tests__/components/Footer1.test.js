// import dependencies
import React from 'react';
// import react dom
import ReactDOM from 'react-dom';
// import react-testing methods
import {render, cleanup, fireEvent} from '@testing-library/react';
// add custom jest matchers from jest-dom
import "@testing-library/jest-dom/extend-expect";
// import renderer for take snapshots
import renderer from 'react-test-renderer';

// import the component for testing
import Footer1 from '../../components/Footer1';


afterEach(cleanup);

it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Footer1 />, div)
})

it("renders footer correctly", () => {
    const {getByTestId} = render(<Footer1 />)
    expect(getByTestId('designer')).toBeTruthy();
    expect(getByTestId('designer')).toHaveTextContent("Designed by University of Moratuwa");
})


it("matches snapshot", () => {
    const tree = renderer.create(<Footer1 />).toJSON();
    expect(tree).toMatchSnapshot();
})