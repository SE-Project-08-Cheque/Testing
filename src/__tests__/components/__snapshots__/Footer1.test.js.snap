// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<nav
  className="css-36cu0e"
>
  <div
    className="css-1autcxr"
  >
    <p
      className="chakra-text css-kbn7if"
      data-testid="designer"
    >
      Designed by University of Moratuwa | 2021 All Rights Reserved
    </p>
  </div>
  <div
    className="css-og5ssx"
  />
</nav>
`;
